.gitlab-ci.yml

```
@@ -64,6 +65,34 @@ test:unit:
     paths:
       - coverage
 
+test:docs-links:
+  stage: test
+  except:
+    - /^(staging|saas-[a-zA-Z0-9.]+)$/
+  before_script:
+    - apk add --no-cache curl
+  script:
+    - links=$(grep -r docs.mender.io src/ | grep -v snapshots | sed -e 's,\${docsVersion},'${DOCS_VERSION}'/,g' | egrep -o 'https?://[^ `]+' | sort | uniq)
+    - error=0
+    - for url in $links; do
+    -   if ! curl --output /dev/null --silent --head --fail $url; then
+    -     error=1
+    -     echo "[ko] $url"
+    -   else
+    -     echo "[ok] $url"
+    -   fi
+    - done
+    - if [ $error -gt 0 ]; then
+    -   exit 1
+    - fi
+
+test:docs-links:hosted:
+  extends: test:docs-links
+  only:
+    - /^(staging|saas-[a-zA-Z0-9.]+)$/
```

We need to blank out `except` here, otherwise GitLab will compose the job with only + except (from the base job) and nothing will run.
```suggestion
  only:
    - /^(staging|saas-[a-zA-Z0-9.]+)$/
  except: []
```

---

GitLab CI/CD is tricky, thanks!